tests/schema/tests.py

```diff
@@ -1791,6 +1793,94 @@ def test_text_field_with_db_index(self):
 
+    @skipUnlessDBFeature('supports_expression_indexes')
+    def test_func_index(self):
+        func_index = Index(fields=[Lower('title').desc()], name='lorem_ipsum_idx')
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.create_model(Book)
+            editor.add_index(Book, func_index)
+            sql = func_index.create_sql(Book, editor)
+
+        sql = sql.upper()
+        self.assertIn('LOWER(', sql)
+        self.assertIn('TITLE', sql)
+        self.assertIn('LOREM_IPSUM_IDX', sql)
+
+        with connection.schema_editor() as editor:
+            sql = func_index.remove_sql(Book, editor)
+            editor.remove_index(Book, func_index)
+            editor.delete_model(Book)
+            editor.delete_model(Author)
+
+        sql = sql.upper()
+        self.assertIn('LOREM_IPSUM_IDX', sql)
+
+    @skipUnlessDBFeature('supports_expression_indexes')
+    def test_func_index_math_expression(self):
+        func_index = Index(fields=[F('height') / (F('weight') + Value(5))], name='lorem_ipsum_idx')
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.add_index(Author, func_index)
+            sql = func_index.create_sql(Author, editor)
+
+        sql = sql.upper()
+        self.assertTrue(
+            sql.index('HEIGHT') <
+            sql.index('/') <
+            sql.index('WEIGHT') <
+            sql.index('+') <
+            sql.index('5')
+        )
+        self.assertIn('LOREM_IPSUM_IDX', sql)
+
+        with connection.schema_editor() as editor:
+            sql = func_index.remove_sql(Author, editor)
+            editor.remove_index(Author, func_index)
+            editor.delete_model(Author)
+
+        sql = sql.upper()
+        self.assertIn('LOREM_IPSUM_IDX', sql)
+
+    @skipUnlessDBFeature('supports_expression_indexes')
+    def test_func_index_invalid_field(self):
+        func_index = Index(fields=[Lower('blub')], name='dolor_idx')
+        with self.assertRaisesMessage(ValueError, "Invalid reference to field 'blub' on model 'schema.Author'"):
+            with connection.schema_editor() as editor:
+                editor.add_index(Author, func_index)
+                func_index.create_sql(Author, editor)
+
+    @skipIfDBFeature('supports_expression_indexes')
+    def test_no_expression_index_support_forbidden(self):
+        indexes = [
+            Index(fields=[Lower('name')], name='dolor_idx'),
+            Index(fields=[Lower('name').desc()], name='dolor_idx'),
+            Index(
+                fields=[OrderBy(OrderBy(Lower(Lower(F('name')), descending=True), descending=False))],
+                name='dolor_idx'
+            ),
+            Index(fields=['height', Lower('name').asc()], name='dolor_idx'),
+        ]
+
+        for index in indexes:
+            with self.subTest(index=index):
+                with self.assertRaisesRegex(ValueError, "Not creating expression index:.*"):
```

Comment:
`assertRaisesRegex` should be avoided, I believe, cc @timgraham 

I've seen a pattern in the migration tests where `self.assertIn` is used to check for parts of the message.